modules/schema/src/main/java/org/apache/ignite/internal/schema/NumericTypeUtils.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.internal.schema;
+
+import java.math.BigDecimal;
+import java.math.BigInteger;
+import org.apache.ignite.schema.ColumnType;
+
+/**
+ * Numeric types utility class.
+ * */
+public class NumericTypeUtils {
+    /**
+     * check precision of BigInteger value.
+     */
+    public static boolean precisionDoesNotFit(BigInteger val, int precision) {
```

Comment:
I think method can be dropped and check can be inlined as
`if (type.precision > 0 && val.precision <= type.precision())`
Can we use zero as magic value for 'unlimited' as it is done for string/bytes?

Seems, zero precision make no sense and forbidden by SQL standard and will be compliant with others varlen types.

Reply:
Yes, we can use zero. I finally removed Utils class and inlined suggested expression.